core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -464,17 +511,45 @@ open class TileInfo {
                     stats.add(unique.stats)
                 }
             }
+        }
 
+        return stats
+    }
+    
+    @Suppress("MemberVisibilityCanBePrivate")
+    fun getImprovementPercentageStats(improvement: TileImprovement, observingCiv: CivilizationInfo, city: CityInfo?): Stats {
+        val stats = Stats()
+        val conditionalState = StateForConditionals(civInfo = observingCiv, cityInfo = city, tile = this)
+        
+        // I would love to make an interface 'canCallMatchingUniques' 
+        // from which both cityInfo and CivilizationInfo derive, so I don't have to duplicate all this code
+        // But something something too much for this PR.
```

Comment:
Double "something".

Reply:
[Intentional](https://www.urbandictionary.com/define.php?term=something%20something) though I agree it's a bit informal, I agree. I can clarify the comment if you want

Reply:
Ouch.. did not know that slang. Keep it in then. But please, do not use in your thesis. :)

Reply:
[ooh I can picture the prof  :eyes:](https://translate.google.com/?source=osdd&sl=auto&tl=nl&text=something%20something%20thesis&op=translate)

Reply:
Duplicate :link: